src/mapblock_mesh.cpp

```diff
@@ -1355,7 +1357,7 @@ bool MapBlockMesh::animate(bool faraway, float time, int crack, u32 daynight_rat
 				i != m_daynight_diffs.end(); i++)
 		{
 			scene::IMeshBuffer *buf = m_mesh->getMeshBuffer(i->first);
-			video::S3DVertex *vertices = (video::S3DVertex*)buf->getVertices();
+			video::S3DVertexTangents *vertices = (video::S3DVertexTangents*)buf->getVertices();
```

Comment:
Tangent space is not just for bump/parallax mapping. Every advanced shader effect will need that data. Because it wasnt aviable before i couldnt continue with any further work on shaders.